### PR TITLE
Add more logging

### PR DESCRIPTION
**Pull request by geerteltink**
|    Q          |   A
|-------------- | ------
| New Feature   | yes

### Description

This PR ads more logging.

### CODE REVIEW

File: test/unit/Changelog/BumpAndCommitChangelogVersionViaKeepAChangelogTest.php [hunk 1]
```diff
@@ -225,29 +228,29 @@ private function createMockChangelog(): string
 
     private const CHANGELOG_STUB = <<< 'CHANGELOG'
         # Changelog
-
```
Comment (weierophinney) [inline test/unit/Changelog/BumpAndCommitChangelogVersionViaKeepAChangelogTest.php:228]: Do the tests that use these continue to work? I'd originally left these spaces in as they get trimmed when the nowdoc is evaluated by the engine, but I'm not sure what happens if they're omitted.

**Review by weierophinney — COMMENTED** [review]
This is a great start, @geerteltink !

I'd also add logging in the various commands, before and/or after major activities (fetching/building changelogs, making API calls, etc.), as that will help us pinpoint where issues are occurring even if the services being called don't have logging integrated.

**Comment by geerteltink** [comment]
I wasn't sure about logging in the command only or the services. I guess both is better.

**Comment by stof** [comment]
When doing work through the API, it would also be great to output the response content when getting an error response from the API instead of only having the failed invariant saying "something went wrong" (or a similar generic message).

**Comment by Ocramius** [comment]
That is an excellent idea: adding logging to the PSR-18 client should suffice, I suppose?

**Comment by Ocramius** [comment]
Closing here: to be retried in a new PR, if needed, but the rebase work is equivalent to the dev work here :)

**Pull request closed without merging**